plugins/bigquery/dbt/include/bigquery/macros/copy_table.sql

```diff
@@ -0,0 +1,31 @@
+{# copy_table macro allows for a table copy job to be submitted to BigQuery, which should run
+ # faster than `select * from`. The macro takes in a BigQueryRelation (returned by a ref() or
+ # source()) defining the source table for the copy, which is forwarded to the adapter. If the
+ # materialization is set to table, creates a table or overwrites and existing one. If incremental,
+ # appends to the existing table. Other materializations are not supported, and throw an error.
+ #}
+{% macro copy_table(source) %}
+
+  {%- if not execute: -%}
+    {%- set materialized_method = config.source_config['config'].get('materialized', '') -%}
```

Comment:
I didn't realize `config.source_config` was even available in the context, but I see now that it is. This doesn't exist in 0.18.x anymore (and never should have been accessible).

I think what you should be going for here is `model.get('config', {}).get('materialized', '')`.

If `model` were always a model, it'd just be `model.config.get('materialized, '')`, but it can be a macro. Maybe macros should get an empty config or something.

Reply:
Makes sense, done.